src/Agent.Worker/ContainerOperationProvider.cs

```diff
@@ -17,8 +17,11 @@ namespace Microsoft.VisualStudio.Services.Agent.Worker
     [ServiceLocator(Default = typeof(ContainerOperationProvider))]
     public interface IContainerOperationProvider : IAgentService
     {
-        Task StartContainerAsync(IExecutionContext executionContext, object data);
```

Comment:
I refactored this into plural Containers. Simplifies the logic of starting the job container + sidecar containers using the same code. 

Made private because StartContainerAsync (no s) should not be called outside of the StartContainerAsyncs wrapper, because the wrapper handles previous job cleanup (so that it only runs once per job)